### ISSUE TITLE
Comments should be improved in all my_blocks

### ISSUE THREAD

**Issue by choikk** [DreamCatchers16105/MDState2017#2]
Some my-blocks have poor comments so it is difficult to follow what is being done at each part of the program.  Please add comments as detail as possible for future reference.

**Comment by SoumyaJailwala** [comment]
Do you mean within the program inside the myblock?

**Comment by choikk** [comment]
Yes, within the program inside the myblock.   You will need to start commenting specially the blocks that I helped you a lot.  It will help you clearly understand the flow of the code.
